src/Swup.ts

```diff
@@ -188,6 +191,14 @@ export default class Swup {
 		// Mount plugins
 		this.options.plugins.forEach((plugin) => this.use(plugin));
 
+		// Install user hooks
+		for (const [key, handler] of Object.entries(this.options.hooks)) {
+			// Buld options object from hook modifier: 'content:replace.before' => { before: true }
+			const [hook, modifiers] = this.hooks.parseName(key as HookName);
```

Comment:
This can be simplified, if we use the suggested type simplification down below. Also, then `@ts-expect-error` is not needed anymore

```js
const [hook, modifiers] = this.hooks.parseName(key as HookName | HookNameWithModifier);
this.hooks.on(hook, handler, modifiers);
```